app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -494,15 +483,12 @@ public void cancelCurrentTask() {
         stateHolder.cancelTask();
     }
 
-    protected void saveLastQueryString(String key) {
-        SharedPreferences settings = getSharedPreferences(CommCarePreferences.ACTIONBAR_PREFS, 0);
-        SharedPreferences.Editor editor = settings.edit();
-        editor.putString(key, lastQueryString);
-        editor.commit();
+    protected void restoreLastQueryString() {
+        lastQueryString = CommCareApplication._().getCurrentSession().getCurrentFrameStepExtra(KEY_LAST_QUERY_STRING);

```

Comment:
Hmmm, looks like that was also the behavior before with
```
lastQueryString = settings.getString(key, null);
```
so we should be good.